src/components/tiles/tile-navigator.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { CSSProperties, useMemo, useRef } from "react";
+import { observer } from "mobx-react-lite";
+import useResizeObserver from "use-resize-observer";
+
+import { ITileProps } from "./tile-component";
+
+import NavigatorMoveIcon from "../../assets/icons/navigator-move-icon.svg";
+
+import "./tile-navigator.scss";
+
+interface INavigatorProps {
+  children: React.ReactElement<ITileProps>;
+}
+
+const navigatorSize = { width: 90, height: 62 };
+
+export const TileNavigator = observer(function TileNavigator({children}: INavigatorProps) {
+  const { height=0, model, tileElt } = children.props;
+  const tileWidth = tileElt?.clientWidth || 0;
+  const tileHeight = tileElt?.clientHeight || 0;
+  const scaleX = navigatorSize.width / tileWidth;
+  const scaleY = navigatorSize.height / height;
+  const scaleFactor = Math.min(scaleX, scaleY);
+  const translate = ((1 - scaleFactor) * .5) * 100;
+  const zoomLevel = "zoom" in model.content ? model.content.zoom : undefined;
+  const displayZoomLevel = typeof zoomLevel === "number" ? `${Math.round(zoomLevel * 100)}%` : "";
+  const containerRef = useRef<HTMLDivElement>(null);
+  const placementButtonRef = useRef<HTMLButtonElement>(null);
+
+  useResizeObserver({ref: tileElt});
+
+  const tileContentStyle: CSSProperties = useMemo(() => {
+    return {
+      width: `${tileWidth}px`,
+      height: `${tileHeight}px`,
+      transform: `translate(${translate}%, ${translate}%) scale(${scaleFactor})`
+    };
+  }, [scaleFactor, tileHeight, tileWidth, translate]);
+
+  const handlePlacementButtonClick = () => {
+    containerRef.current?.classList.add("animate");
+    containerRef.current?.classList.toggle("top");
+    placementButtonRef.current?.classList.toggle("top");
+  };
+
+  return (
+    <div ref={containerRef} className="tile-navigator-container" data-testid="tile-navigator-container">
+      <div className="tile-navigator" data-testid="tile-navigator">
+        <div className="tile-navigator-content-area">
+          <div className="tile-navigator-tile-content" style={tileContentStyle}>
+            {React.cloneElement(children, { readOnly: true, navigatorAllowed: false })}
```

Comment:
I had not encountered `cloneElement` before.  The [docs](https://react.dev/reference/react/cloneElement) suggest that there may be better alternatives in most cases.  I wonder if the caller could just specify those props rather than adding them here?

Reply:
Thanks. I was hoping to remove any responsibility on the part of the caller for setting those props to the safe values. I'll look into the alternatives suggested in the documentation. I suppose it would be OK to just have the caller specify the values, but I was hoping the tile navigator component itself could simply ensure they were always set to the right thing.